app/models/competition_tab.rb

```diff
@@ -34,6 +34,16 @@ def slug
     competition.tabs.where("display_order > ?", display_order).update_all("display_order = display_order - 1")
   end
 
+  validate :verify_if_full_urls, on: :update
```

Comment:
Why is this limited to `on: :update`? If a new tab is created, this would currently allow relative URLs to "slip through".

Reply:
I didn't think about that case. I just tried cloning a competition that has this error, the result was the competition got created, but the tab didn't get created and no error was thrown to the delegate. Will it be better if we allow to do this mistake during clone and stop them only during update?

Reply:
That's... odd :sweat_smile:
I mean, in the current scenario with your code as-is right now, of course the error won't get thrown to the Delegate. Because cloning makes it so that a _clone_ of everything is _created_, meaning the `on: :update` wouldn't trigger during the cloning process.

Reply:
I mean if I add `:create`, still the error won't be thrown to delegate. The cloning will happen without any issues, just that the tabs with error won't get created which might potentially create confusions among delegates why those didn't get created.

Reply:
Currently since it's just `:update`, the faulty tabs are also getting created, so won't create any confusions to delegate.

Reply:
Okay, we definitely need to look into why that happens. Because you just tested it with cloning. But if a Delegate is legitimately creating a _fully new_ competition tab, then they would also be able to circumvent this filter, making it useless in at least ~30% of cases.